pkg/trace/trace.go

```diff
@@ -12,59 +13,66 @@ import (
 	"time"
 )
 
-func NewTracer(sporingsURL string, context string) *Tracer {
-	return &Tracer{
-		url:     sporingsURL,
-		context: context,
-		enabled: sporingsURL != "" && context != "",
+type Trace interface {
+	AddImageMetadata(data interface{}) error
+	AddBaseImageMetadata(application config.ApplicationSpec, imageInfo *runtime.ImageInfo, containerConfig *docker.ContainerConfig)
+}
+
+func NewTraceClient(sporingURL string) Trace {
+	return &TraceClient{
+		url:     sporingURL,
+		enabled: sporingURL != "",
 	}
 }
 
-type Tracer struct {
+type TraceClient struct {
 	url     string
-	context string
 	enabled bool
 }
 
-func (t *Tracer) AddImageMetadata(data interface{}) {
-	if t.enabled {
-		ctx := context.Background()
-		timeoutIn := time.Now().Add(5 * time.Second)
-		ctx, cancelFunc := context.WithDeadline(ctx, timeoutIn)
-		defer cancelFunc()
+func (traceClient *TraceClient) AddImageMetadata(data interface{}) error {
+	ctx := context.Background()
+	timeoutIn := time.Now().Add(5 * time.Second)
+	ctx, cancelFunc := context.WithDeadline(ctx, timeoutIn)
+	defer cancelFunc()
```

Comment:
Skal man ikke skrive "defer cancelFunc", og ikke kalle metoden? Men sikkert riktig, jeg bare stusser...

Reply:
tror den biten var der fra før. 
Ser riktig ut for meg :  https://yourbasic.org/golang/defer/

Reply:
@bj0rn hva tenker du om denne ?

Reply:
Du har helt rett, @K-Fl. "The deferred call’s arguments are evaluated immediately, even though the function call is not executed until the surrounding function returns.".